pkg/kmm/containers_test.go

```diff
@@ -0,0 +1,230 @@
+			args:               []string{"arg"},
+			rawargs:            []string{"rawarg1", "rawargs2"},
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := NewModLoaderContainerBuilder("test")
+		testBuilder.WithModprobeSpec(testCase.dirName, testCase.fwPath,
+			testCase.parameters, testCase.args, testCase.rawargs, testCase.moduleLoadingOrder)
+
+		assert.Equal(t, testCase.dirName, testBuilder.definition.Modprobe.DirName)
+		assert.Equal(t, testCase.fwPath, testBuilder.definition.Modprobe.FirmwarePath)
+		assert.Equal(t, testCase.parameters, testBuilder.definition.Modprobe.Parameters)
+		assert.Equal(t, testCase.moduleLoadingOrder, testBuilder.definition.Modprobe.ModulesLoadingOrder)
+
+		if len(testCase.args) > 0 {
+			assert.Equal(t, testCase.args, testBuilder.definition.Modprobe.Args.Load)
+		}
+
+		if len(testCase.rawargs) > 0 {
+			assert.Equal(t, testCase.rawargs, testBuilder.definition.Modprobe.RawArgs.Load)
+		}
+	}
+}
+
+func TestModuleLoaderContainerBuilder_WithImagePullPolicy(t *testing.T) {
+	testCases := []struct {
+		imagePolicy    string
+		expectedPolicy string
+		expectedError  string
+	}{
+		{
+			imagePolicy:    "",
+			expectedPolicy: "IfNotPresent",
+			expectedError:  "'policy' can not be empty",
+		},
+		{
+			imagePolicy:    "Never",
+			expectedPolicy: "Never",
+			expectedError:  "",
+		},
+		{
+			imagePolicy:    "Always",
+			expectedPolicy: "Always",
+			expectedError:  "",
+		},
+		{
+			imagePolicy:    "IfNotPresent",
+			expectedPolicy: "IfNotPresent",
+			expectedError:  "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := NewModLoaderContainerBuilder("test")
+		testBuilder.WithImagePullPolicy(testCase.imagePolicy)
+
+		assert.Equal(t, testCase.expectedError, testBuilder.errorMsg)
+
+		if testCase.expectedError == "" {
+			assert.Equal(t, v1.ImagePullPolicy(testCase.expectedPolicy), testBuilder.definition.ImagePullPolicy)
+		}
+	}
+}
+
+func TestModuleLoaderContainerBuilder_WithKernelMapping(t *testing.T) {
+	testCases := []struct {
+		mapping       *v1beta1.KernelMapping
+		expectedError string
+	}{
+		{
+			mapping:       buildRegExKernelMapping(""),
+			expectedError: "'mapping' can not be empty nil",
+		},
+		{
+			mapping:       buildRegExKernelMapping("^.+$"),
+			expectedError: "",
+		},
+		{
+			mapping:       buildLiteralKernelMapping("5.14.0-70.58.1.el9_0.x86_64"),
+			expectedError: "",
+		},
+		{
+			mapping:       buildLiteralKernelMapping(""),
+			expectedError: "'mapping' can not be empty nil",
+		},
+	}
+
+	for _, testcase := range testCases {
+		testBuilder := NewModLoaderContainerBuilder("test")
+		testBuilder.WithKernelMapping(testcase.mapping)
+
+		if testcase.expectedError != "" {
+			assert.Equal(t, testBuilder.errorMsg, testcase.expectedError)
+		} else {
+			assert.Equal(t, testBuilder.definition.KernelMappings[0], *testcase.mapping)
+		}
+	}
+}
+
+func TestModuleLoaderContainerBuilder_WithOptions(t *testing.T) {
```

Comment:
Please remove underscore. 
It's easy to add 2 test cases here:

```
	testBuilder := NewModLoaderContainerBuilder("test").WithOptions(
		func(builder *ModuleLoaderContainerBuilder) (*ModuleLoaderContainerBuilder, error) {
			return builder, nil
		})

	testBuilder = NewModLoaderContainerBuilder("test").WithOptions(
		func(builder *ModuleLoaderContainerBuilder) (*ModuleLoaderContainerBuilder, error) {
			return builder, fmt.Errorf("error")
		})
	assert.Equal(t, "error", testBuilder.errorMsg)
```

Reply:
Done . Thanks!